File: ChatApp/app/src/main/java/top/chilfish/chatapp/adapter/ContactAdapter.java
package top.chilfish.chatapp.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.List;

import top.chilfish.chatapp.databinding.ItemContactBinding;
import top.chilfish.chatapp.entity.Profile;

public class ContactAdapter extends BaseAdapter<Profile> {
  private List<Profile> mContactList;

  public ContactAdapter(List<Profile> dataList) {
    super(dataList);
    mContactList = dataList;
  }

  @NonNull
  @Override
  public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    ItemContactBinding binding = ItemContactBinding.inflate(LayoutInflater.from(parent.getContext()),
        parent, false);

    return new ContactViewHolder(binding);
  }

  public class ContactViewHolder extends BaseAdapter<Profile>.ViewHolder {
    private ItemContactBinding binding;

    public ContactViewHolder(ItemContactBinding binding) {
      super(binding.getRoot());
      this.binding = binding;
    }

    @Override
    public void bindData(Profile data) {
      try {
        Glide.with(itemView.getContext())
            .load(data.getAvatar())
            .into(binding.contactAvatar);

        binding.contactName.setText(data.getName());
      } catch (Exception e) {
        e.printStackTrace();
      }
    }
  }
}

File: ChatApp/app/src/main/java/top/chilfish/chatapp/adapter/ContactAdapter.kt
package top.chilfish.chatapp.adapter

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.core.content.ContextCompat.startActivity
import com.bumptech.glide.Glide
import top.chilfish.chatapp.R
import top.chilfish.chatapp.databinding.ItemContactBinding
import top.chilfish.chatapp.entity.Profile
import top.chilfish.chatapp.ui.activities.ProfileActivity

class ContactAdapter : BaseAdapter<Profile, ItemContactBinding>() {
    override fun getLayoutId(): Int {
        return R.layout.item_contact
    }

    override fun onBindViewHolder(holder: ViewHolder<ItemContactBinding>, position: Int) {
        super.onBindViewHolder(holder, position)
        val data = items[position]
        val binding = holder.binding
        try {
            Glide.with(holder.itemView.context)
                .load(data.avatar)
                .into(binding.contactAvatar)

            binding.contactName.text = data.name
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onItemClicked(item: Profile, view: View) {
        val bundle = Bundle()
        bundle.putSerializable("profile", item)
        val intent = Intent(view.context, ProfileActivity::class.java)
        intent.putExtras(bundle)
        startActivity(view.context, intent, null)
    }
}